File: python/datastructure/otterop/datastructure/test_list.py
from otterop.lang.array import Array as _Array
from otterop.lang.generic import Generic as _Generic
from otterop.lang.string import String as _String
from otterop.test.test_base import TestBase as _TestBase
from otterop.datastructure.list import List as _List

class TestList(_TestBase):

    def add(self):
        l = _List()
        l.add(_String.wrap("a"))
        l.add(_String.wrap("b"))
        l.add(_String.wrap("c"))
        l.add(_String.wrap("d"))
        l.add(_String.wrap("e"))
        self.assert_true(l.size() == 5, _String.wrap("Size should be 5"))

    def add_range(self):
        generic_t = _Generic()
        generic_t_zero = generic_t.zero()
        l = _List()
        to_add = _Array.new_array(5, generic_t_zero)
        to_add.set(0, _String.wrap("a"))
        to_add.set(1, _String.wrap("b"))
        to_add.set(2, _String.wrap("c"))
        to_add.set(3, _String.wrap("d"))
        to_add.set(4, _String.wrap("e"))
        l.add_array(to_add)
        self.assert_true(l.size() == 5, _String.wrap("Size should be 5"))

    def remove_index(self):
        
        l = _List()
        l.add(_String.wrap("a"))
        l.add(_String.wrap("b"))
        l.add(_String.wrap("c"))
        l.add(_String.wrap("d"))
        l.add(_String.wrap("e"))
        l.remove_index(1)
        l.remove_index(1)
        self.assert_true(l.size() == 3, _String.wrap("Size should be 3"))
        val = l.get(0)
        self.assert_true(val.compare_to(_String.wrap("a")) == 0, _String.wrap("First element should be a"))
        val = l.get(1)
        self.assert_true(val.compare_to(_String.wrap("d")) == 0, _String.wrap("Second element should be d"))
        val = l.get(2)
        self.assert_true(val.compare_to(_String.wrap("e")) == 0, _String.wrap("Third element should be e"))

    def remove_range(self):
        
        l = _List()
        l.add(_String.wrap("a"))
        l.add(_String.wrap("b"))
        l.add(_String.wrap("c"))
        l.add(_String.wrap("d"))
        l.add(_String.wrap("e"))
        l.remove_range(3, 2)
        self.assert_true(l.size() == 3, _String.wrap("Size should be 3"))
        val = l.get(0)
        self.assert_true(val.compare_to(_String.wrap("a")) == 0, _String.wrap("First element should be a"))
        val = l.get(1)
        self.assert_true(val.compare_to(_String.wrap("b")) == 0, _String.wrap("Second element should be b"))
        val = l.get(2)
        self.assert_true(val.compare_to(_String.wrap("c")) == 0, _String.wrap("Third element should be c"))

def test_add():
    TestList().add()

def test_add_range():
    TestList().add_range()

def test_remove_index():
    TestList().remove_index()

def test_remove_range():
    TestList().remove_range()

File: dotnet/tests/Datastructure/Otterop/Datastructure/TestList.cs
namespace Otterop.Datastructure
{
    public class TestList : Otterop.Test.TestBase
    {
        [Otterop.Test.Test]
        public void Add()
        {
            List<Otterop.Lang.String> l = new List<Otterop.Lang.String>();
            l.Add(Otterop.Lang.String.Wrap("a"));
            l.Add(Otterop.Lang.String.Wrap("b"));
            l.Add(Otterop.Lang.String.Wrap("c"));
            l.Add(Otterop.Lang.String.Wrap("d"));
            l.Add(Otterop.Lang.String.Wrap("e"));
            AssertTrue(l.Size() == 5, Otterop.Lang.String.Wrap("Size should be 5"));
        }

        [Otterop.Test.Test]
        public void AddRange()
        {
            Otterop.Lang.Generic<Otterop.Lang.String> genericT = new Otterop.Lang.Generic<Otterop.Lang.String>();
            Otterop.Lang.String genericTZero = genericT.Zero();
            List<Otterop.Lang.String> l = new List<Otterop.Lang.String>();
            Otterop.Lang.Array<Otterop.Lang.String> toAdd = Otterop.Lang.Array.NewArray(5, genericTZero);
            toAdd.Set(0, Otterop.Lang.String.Wrap("a"));
            toAdd.Set(1, Otterop.Lang.String.Wrap("b"));
            toAdd.Set(2, Otterop.Lang.String.Wrap("c"));
            toAdd.Set(3, Otterop.Lang.String.Wrap("d"));
            toAdd.Set(4, Otterop.Lang.String.Wrap("e"));
            l.AddArray(toAdd);
            AssertTrue(l.Size() == 5, Otterop.Lang.String.Wrap("Size should be 5"));
        }

        [Otterop.Test.Test]
        public void RemoveIndex()
        {
            Otterop.Lang.String val;
            List<Otterop.Lang.String> l = new List<Otterop.Lang.String>();
            l.Add(Otterop.Lang.String.Wrap("a"));
            l.Add(Otterop.Lang.String.Wrap("b"));
            l.Add(Otterop.Lang.String.Wrap("c"));
            l.Add(Otterop.Lang.String.Wrap("d"));
            l.Add(Otterop.Lang.String.Wrap("e"));
            l.RemoveIndex(1);
            l.RemoveIndex(1);
            AssertTrue(l.Size() == 3, Otterop.Lang.String.Wrap("Size should be 3"));
            val = l.Get(0);
            AssertTrue(val.CompareTo(Otterop.Lang.String.Wrap("a")) == 0, Otterop.Lang.String.Wrap("First element should be a"));
            val = l.Get(1);
            AssertTrue(val.CompareTo(Otterop.Lang.String.Wrap("d")) == 0, Otterop.Lang.String.Wrap("Second element should be d"));
            val = l.Get(2);
            AssertTrue(val.CompareTo(Otterop.Lang.String.Wrap("e")) == 0, Otterop.Lang.String.Wrap("Third element should be e"));
        }

        [Otterop.Test.Test]
        public void RemoveRange()
        {
            Otterop.Lang.String val;
            List<Otterop.Lang.String> l = new List<Otterop.Lang.String>();
            l.Add(Otterop.Lang.String.Wrap("a"));
            l.Add(Otterop.Lang.String.Wrap("b"));
            l.Add(Otterop.Lang.String.Wrap("c"));
            l.Add(Otterop.Lang.String.Wrap("d"));
            l.Add(Otterop.Lang.String.Wrap("e"));
            l.RemoveRange(3, 2);
            AssertTrue(l.Size() == 3, Otterop.Lang.String.Wrap("Size should be 3"));
            val = l.Get(0);
            AssertTrue(val.CompareTo(Otterop.Lang.String.Wrap("a")) == 0, Otterop.Lang.String.Wrap("First element should be a"));
            val = l.Get(1);
            AssertTrue(val.CompareTo(Otterop.Lang.String.Wrap("b")) == 0, Otterop.Lang.String.Wrap("Second element should be b"));
            val = l.Get(2);
            AssertTrue(val.CompareTo(Otterop.Lang.String.Wrap("c")) == 0, Otterop.Lang.String.Wrap("Third element should be c"));
        }

    }

}
